policies.go

```diff
@@ -880,6 +880,45 @@ func (d *dcAwareRR) Pick(q ExecutableQuery) NextHost {
 	return roundRobbin(int(nextStartOffset), d.localHosts.get(), d.remoteHosts.get())
 }
 
+// ReadyPolicy defines a policy for when a HostSelectionPolicy can be used. After
+// each host connects during session initialization, the Ready method will be
+// called. If you only need a single Host to be up you can wrap a
+// HostSelectionPolicy policy with SingleHostReadyPolicy.
+type ReadyPolicy interface {
+	Ready() bool
+}
+
+// SingleHostReadyPolicy wraps a HostSelectionPolicy and returns Ready after a
+// single host has been added via AddHost
```

Comment:
I think this policy should wait for `HostUp`, not `AddHost`. It's not described in `HostStateNotifier` interface, but I believe `AddHost` and `RemoveHost` are for adding hosts (even down ones) and `HostUp`/`HostDown` are for notifying of host status changes. Even though most policies don't distinguish those methods, I think for ReadyPolicy it makes sense to take advantage of this difference. When using `HostUp`, we should be able to add (or bulk add) the hosts before starting to connect anywhere and then update the policy with HostUp as the connections go up.

Reply:
I couldn't find a single thing calling `HostUp` so I didn't bother putting it in there. Where is that being called?